nativelib/src/main/resources/scala-native/gc/shared/Parsing.c

```diff
@@ -50,17 +54,52 @@ size_t Parse_Size_Or_Default(const char *str, size_t defaultSizeInBytes) {
     return defaultSizeInBytes;
 }
 
+const char *get_defined_or_env(const char *envName) {
+    if (envName == NULL)
+        return NULL;
+
+    const char *env = getenv(envName);
+    if (env != NULL)
+        return env; // override with env var
```

Comment:
There is always the discussion about when and where arguments get validated.   Earlier usually allows more informative error messages.

I think all of the expected valid values are expected to be non-negative
integers, where the integer format is defined by the os.

If the intention is to defer argument testing  to each gc, because
each gc had different requirements, it might be worthwhile to
say that in a comment.  Elsewise folks who assume all user
input is invalid will be left wondering.

Reply:
Yes, the null checks are a bit strange. This one is because of no env var set but I think some asserts would be better in some places. You would have to help me with basic C stuff but the error handling - user validation is pretty much non-existent here. Even a warning to stdout for bad input might be good. For another day I am afraid.